src/command_frame.cc

```diff
@@ -1,71 +1,98 @@
-                                 resolver(options))
-      .wait();
-
-  for (const auto &[key, entry] : frame) {
-    std::cout << "(LOCATION) URI: ";
-    std::cout << key.second << "\n";
-    std::cout << "    Schema           : " << entry.root.value_or("<ANONYMOUS>")
-              << "\n";
-    std::cout << "    Pointer          :";
-    if (!entry.pointer.empty()) {
-      std::cout << " ";
-    }
+// Function to pad each line of a string except the first line
+std::string padLines(const std::string &input, int padding) {
+    std::istringstream inputStream(input);
+    std::ostringstream outputStream;
+    std::string line;
+    std::string pad(padding, ' ');
 
-    sourcemeta::jsontoolkit::stringify(entry.pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Base URI         : " << entry.base << "\n";
-    std::cout << "    Relative Pointer :";
-    if (!entry.relative_pointer.empty()) {
-      std::cout << " ";
+    // Handle the first line
+    if (std::getline(inputStream, line)) {
+        outputStream << line << "\n";
     }
 
-    sourcemeta::jsontoolkit::stringify(entry.relative_pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Dialect          : " << entry.dialect << "\n";
-  }
-
-  for (const auto &[pointer, entry] : references) {
-    std::cout << "(REFERENCE) URI: ";
-    sourcemeta::jsontoolkit::stringify(pointer.second, std::cout);
-    std::cout << "\n";
-
-    std::cout << "    Type             : ";
-    if (pointer.first == sourcemeta::jsontoolkit::ReferenceType::Dynamic) {
-      std::cout << "Dynamic";
-    } else {
-      std::cout << "Static";
+    while (std::getline(inputStream, line)) {
+        outputStream << pad << line << "\n";
     }
-    std::cout << "\n";
-    std::cout << "    Destination      : " << entry.destination << "\n";
 
-    if (entry.base.has_value()) {
-      std::cout << "    - (w/o fragment) : " << entry.base.value() << "\n";
+    return outputStream.str();
+}
+
+auto intelligence::jsonschema::cli::frame(const std::span<const std::string> &arguments) -> int {
+    const auto options{parse_options(arguments, {"json", "j"})};
+    CLI_ENSURE(!options.at("").empty(), "You must pass a JSON Schema as input")
+    const sourcemeta::jsontoolkit::JSON schema{
+        sourcemeta::jsontoolkit::from_file(options.at("").front())
+    };
+
+    sourcemeta::jsontoolkit::ReferenceFrame frame;
+    sourcemeta::jsontoolkit::ReferenceMap references;
+    sourcemeta::jsontoolkit::frame(schema, frame, references,
+                                   sourcemeta::jsontoolkit::default_schema_walker,
+                                   resolver(options))
+        .wait();
+
+    bool outputJson = options.contains("json") || options.contains("-j");
+
+    for (const auto &[key, entry] : frame) {
+        std::cout << "(LOCATION) URI: ";
+        std::cout << key.second << "\n";
+        std::cout << "    Schema           : " << entry.root.value_or("<ANONYMOUS>")
```

Comment:
i.e. all of these things you should be transforming to JSON. In the end you want the command to print only one big JSON document with all the information at once

Reply:
Ohh, I got it. I thought it might be this way, just in sometime...